src/test/java/com/eighthlight/fabrial/test/server/AppAcceptanceTest.java

```diff
@@ -3,74 +3,86 @@
 import com.eighthlight.fabrial.http.HttpVersion;
 import com.eighthlight.fabrial.http.Method;
 import com.eighthlight.fabrial.http.RequestBuilder;
-import com.eighthlight.fabrial.server.ServerConfig;
+import com.eighthlight.fabrial.test.file.TempDirectoryFixture;
+import com.eighthlight.fabrial.test.file.TempFileFixtures;
+import com.eighthlight.fabrial.test.http.AppProcessFixture;
 import com.eighthlight.fabrial.test.http.RequestWriter;
 import com.eighthlight.fabrial.test.http.TcpClientFixture;
-import com.eighthlight.fabrial.test.http.TempFileFixture;
-import org.junit.jupiter.api.AfterEach;
-import org.junit.jupiter.api.BeforeEach;
+import com.eighthlight.fabrial.utils.Result;
 import org.junit.jupiter.api.Tag;
 import org.junit.jupiter.api.Test;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.io.BufferedReader;
-import java.io.IOException;
-import java.io.InputStreamReader;
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
 import java.nio.file.Paths;
 
 import static org.hamcrest.CoreMatchers.is;
 import static org.hamcrest.MatcherAssert.assertThat;
 
 @Tag("acceptance")
 public class AppAcceptanceTest {
-  static final Logger logger = LoggerFactory.getLogger(AppAcceptanceTest.class.getName());
-
-  Process appProcess;
-
-  @BeforeEach
-  void setUp() throws IOException {
-    // hard-coded to SNAPSHOT version. might need to fix this eventually...
-    appProcess =
-        new ProcessBuilder(
-            "java",
-            // disable logback error messages (no logstash here)
-            "-Dlogback.statusListenerClass=ch.qos.logback.core.status.NopStatusListener",
-            "-jar", "./build/libs/fabrial-all-1.0-SNAPSHOT.jar")
-            .inheritIO()
-            .start();
-  }
-
-  @AfterEach
-  void tearDown() {
-    appProcess.destroy();
-  }
+  private static final Logger logger = LoggerFactory.getLogger(AppAcceptanceTest.class);
 
   @Test
   void clientConnectsToAppServer() throws IOException {
-    try (TcpClientFixture clientFixture =
-        new TcpClientFixture(ServerConfig.DEFAULT_PORT)) {
+    int testPort = 8081;
+    try (AppProcessFixture appFixture = new AppProcessFixture(testPort , null);
```

Comment:
@trptcolin you might appreciate my leveraging `AutoCloseable` to create reusable fixtures which automatically clean up at the end of the enclosing scope. Maybe not as elegant as an annotation, but still a step in the right direction (away from inheritance)

Reply:
🎉yay composition! 🎉